main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -514,53 +501,66 @@ object Safety {
         acc + (signature -> m)
     }
   }
+  
+  /**
+    * Convert a `java.lang.reflect.Method` to a MethodSignature.
+    */
+  private def getJavaMethodSignature(method: java.lang.reflect.Method) = {
+    MethodSignature(method.getName(), 
+      getFlixType(method.getReturnType),
+      method.getParameterTypes().toList.map(getFlixType))
+  }
+
+  /**
+    * Given a class, get sets representing both the methods that can be implemented,
+    * and the methods that must be implemented.
+    */
+  private def getJavaMethods(clazz: java.lang.Class[_]): (Set[MethodSignature], Set[MethodSignature]) = {
+    val methods = clazz.getMethods().toList.filterNot(m => java.lang.reflect.Modifier.isStatic(m.getModifiers()))
+    val mustImplement = if (clazz.isInterface()) {
+        methods.filterNot(_.isDefault())
+      } else {
+        methods.filter(m => java.lang.reflect.Modifier.isAbstract(m.getModifiers()))
+      }
+    (methods.map(getJavaMethodSignature).toSet, mustImplement.map(getJavaMethodSignature).toSet)
```

Comment:
I am bit concerned about the `toSet`. Could it be expensive and is it necessary?

Reply:
This may be where my rusty Scala knowledge is letting me down. Is there an equivalent of `Set.diff` that works on sequences?

Reply:
Ah okay. You need it for diff. Hmm. Maybe its okay then.